File: solutions/cpp/Problem136.cpp
#pragma once
/*
136. Single Number

Given a non - empty array of integers, every element appears twice except for one. Find that single one.

Note:
Your algorithm should have a linear runtime complexity.Could you implement it without using extra memory ?
*/
#include <vector>


class Solution136 {
public:
	int singleNumber(std::vector<int>& nums) {
		int num = 0;
		for (auto & n : nums)
		{
			num ^= n;
		}
		return num;
	}
	void test() {

	}
}; 

File: solutions/python3/problem136.py
# -*- coding: utf-8 -*-
"""
136. Single Number

Given a non-empty array of integers nums, every element appears twice except for one. Find that single one.

Follow up: Could you implement a solution with a linear runtime complexity and without using extra memory?

Constraints:

1 <= nums.length <= 3 * 104
-3 * 104 <= nums[i] <= 3 * 104
Each element in the array appears twice except for one element which appears only once.
"""


class Solution:
    def singleNumber(self, nums):
        res = 0
        for num in nums:
            res ^= num

        return res
